File: task2.cpp
#include <iostream>

using namespace std;

string task2(const string& input) {
    string output;

    for (const char c : input) {
        switch(c) {
            case '&': output += "&amp;"; break;
            case '<': output += "&lt;"; break;
            case '>': output += "&gt;"; break;
            default: output += c;
        }
    }

    return output;
}

int main() {
    cout << task2("Hello & <World>!") << "\n";
    return 0;
}

File: task2.rs
fn task2(input: &str) -> String {
    let mut output = String::new();

    for c in input.chars() {
        match c {
            '&' => output.push_str("&amp;"),
            '<' => output.push_str("&lt;"),
            '>' => output.push_str("&gt;"),
            _ => output.push(c),
        }
    }

    output
}

fn main() {
    let output = task2("Hello & <World>!");
    println!("{}", output);
}